core/src/conn.rs

```diff
@@ -158,13 +198,37 @@ impl MongoConnection {
                 .build()
                 .unwrap()
         }));
+
         user_options.client_options.connect_timeout =
             login_timeout.map(|to| Duration::new(u64::from(to), 0));
+
         let uuid_repr = user_options.uuid_representation;
-        let (client, runtime) = Self::get_client_and_runtime(user_options, runtime)?;
 
         load_mongosqltranslate_library();
 
+        let is_libmongosqltranslate_compatible_with_driver_version =
+            if get_mongosqltranslate_library().is_some() {
+                let libmongosqltranslate_version = Self::get_libmongosqltranslate_version()?;
+
+                // This appends "|libmongosqltranslate+<version>" to the app_name
+                user_options
+                    .client_options
+                    .app_name
+                    .as_mut()
+                    .expect("app_name unexpectedly has the value `None`")
```

Comment:
I used `expect` here because its not possible for `app_name` to be `None` at this point in the code. You can think of this `expect` as a `unreachable!()`. Also, I feel this way about the few `expects` remaining in this PR. I'm not sure we want to error where they are because they indicate that something is very wrong internally if they are ever hit.

Reply:
Consider what the user will see and what will be logged. It's much better to handle all of these in one place and put an error on it that we could bubble up to inform the user something is horribly wrong and they should report it.

Reply:
Makes sense; I'll change it to an error. Something I just thought of though: should I change my two instances of `unreachable!()` into an error too? Because I feel like the same logic applies there. I'm guessing in general we would rather not panic?

```
            MongoClusterType::Community | MongoClusterType::UnknownTarget => {
                // On connection, these types should get caught and throw an error.
                unreachable!()
            }
```

Reply:
Eventually it should probably panic, but in a centralized, controlled way. We can chat on Slack if you'd like about how to do this. If we go with structs for our commands and implement custom serializer and deserializer we can handle it in one spot.